File: solutions/cpp/log-levels/1/log_levels.cpp
#include <string>

namespace log_line {
std::string message(std::string line) {
    auto pos = line.find(": ");
    return line.substr(pos + 2);
}


std::string log_level(std::string line) {

    // find the brackets
    auto start = line.find('[');
    auto end = line.find(']');

    // extract the substring
    std::string level = line.substr(start + 1, end - start - 1);
    
    return level;
}

std::string reformat(std::string line) {
    // return the reformatted message
    auto start = line.find('[');
    auto end = line.find(']');

    std::string level = line.substr(start+1, end-start-1);

    auto pos = line.find(": ");
    std::string message = line.substr(pos + 2) + " (" + level + ")";
    
    return message;
}
}  // namespace log_line

File: solutions/csharp/log-levels/1/LogLevels.cs
static class LogLine
{
    public static string Message(string logLine)
    {
        int index = logLine.IndexOf(' ');
		return logLine.Substring(index).Trim();
    }

    public static string LogLevel(string logLine)
    {
		int firstIndex = logLine.IndexOf('[');
		int lastIndex = logLine.IndexOf("]");
		return logLine.Substring(firstIndex + 1, lastIndex - 1).ToLower().Trim();
	}

	public static string Reformat(string logLine)
    {
		int firstIndex = logLine.IndexOf("[") + 1;
		int lastIndex = logLine.IndexOf("]") - 1;
		int index = logLine.IndexOf(' ');
		return $"{logLine.Substring(index).Trim()} ({logLine.Substring(firstIndex,lastIndex).ToLower()})";
	}
}
